kafka-streams-quickstart/producer/src/main/java/org/acme/kafka/streams/producer/generator/ValuesGenerator.java

```diff
@@ -60,13 +60,13 @@
     }
 
     @Outgoing("weather-stations")
-    public Flowable<KafkaRecord<Integer, String>> weatherStations() {
+    public Multi<KafkaRecord<Integer, String>> weatherStations() {
         List<KafkaRecord<Integer, String>> stationsAsJson = stations.stream()
```

Comment:
Minor - 2 things:

* we can create a Multi directly from a Stream (`Multi.createFrom().items(stream)`), it will iterate according to demands, which is often better
* I don't believe we need to use `KafkaRecord`, we can use `Record` directly (depends who consume the stream).

Can you check with:

```java
public Multi<Record<Integer, String>> weatherStations() {
   return Multi.createFrom().items(stations.stream()
      .map(s -> Record.of(s.id, "{ \"id\" : " + s.id + ", \"name\" : \"" + s.name + "\" }"))
   );
}
```

Reply:
@cescoffier, could you please let me know which `Record` you mean in this case? Did not find this class right away.

Reply:
This one: `io.smallrye.reactive.messaging.kafka.Record`. It's quite new, so be sure you have the latest Quarkus _master_.